OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/gpx/GpxUtilities.kt

```diff
@@ -1164,6 +1179,11 @@ object GpxUtilities {
 									}
 
 									"link" -> parse.link = parser.getAttributeValue("", "href")
+									"text" -> {
+										if (insideTagDepth["link"]!! > 0) {
```

Comment:
Introduce class `Link` instead of `insideTagDepth`

Reply:
done